src/main/groovy/cruise/umple/tasks/UmpleGenerateTask.groovy

```diff
@@ -4,20 +4,40 @@ import cruise.umple.UmpleConsoleConfig
 import cruise.umple.UmpleConsoleMain
 import cruise.umple.UmpleLanguage
 import cruise.umple.internal.tasks.DefaultUmpleOptions
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.SourceTask
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.*
 import org.gradle.api.tasks.incremental.IncrementalTaskInputs
 
 import static com.google.common.base.Preconditions.checkNotNull
 /**
  * Created by kevin on 15/03/2017.
  */
 class UmpleGenerateTask extends SourceTask {
+    @Input
     private UmpleOptions compileConfig //this contains the DefaultUmpleOption we create within the apply method in UmpleGradlePlugin
+    @OutputFiles
+    private List<File> outputDirs
 
     UmpleGenerateTask() {
-        compileConfig = (DefaultUmpleOptions)(project.extensions.getByName(UmpleOptions.NAME))
+        this.compileConfig = (DefaultUmpleOptions)(project.extensions.getByName(UmpleOptions.NAME))
+        setOutputDirs(this.compileConfig)
+    }
+
+    @OutputFiles
+    void setOutputDirs(UmpleOptions newOpts) {
+        this.outputDirs = new ArrayList<File>()
+
+        UmpleOptions opts = newOpts
```

Comment:
I used the logic in `execute` as the basis for the logic here. I noticed that in `execute` we make a copy of the UmpleOptions (in `setOutputDirs` we do `UmpleOptions opts = newOpts`). Is this copy necessary? I don't think we're changing the `UmpleOption` object.

Reply:
99% sure that's referential copy so it does _literally_ nothing.